x-pack/plugins/snapshot_restore/public/app/sections/home/repository_list/repository_details/repository_details.tsx

```diff
@@ -147,6 +149,30 @@ const RepositoryDetailsUi: React.FunctionComponent<Props> = ({
           </EuiFlexItem>
         </EuiFlexGroup>
         <EuiSpacer size="l" />
+        <EuiTitle size="s">
+          <h3>
+            <FormattedMessage
+              id="xpack.snapshotRestore.repositoryDetails.snapshotsTitle"
+              defaultMessage="Snapshots"
+            />
+          </h3>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        {Number.isInteger(snapshots.count) ? (
```

Comment:
I know it's very unlikely this will happen, but `Number.isInteger(0)` evaluates to `true`, so maybe we should should change this condition to this:

```js
Number(snapshots.count) > 0
```

This will return true for `undefined` and `null` and will work if `count` is provided as a string.

Reply:
ah, this was actually done intentionally. sorry, I realize now that it is not very obvious.

if there was an issue with retrieving snapshot information (such as when there is an error on the repository), the API will return `null`. otherwise if there were no issues with retrieving information and there are no snapshots, it will return `0`

`0 snapshots found` means empty repository
`No snapshot information` means there was an issue with accessing the repository snapshot info

I'm open to copy suggestions for this scenario!

Reply:
Gotcha! Here's my suggestion:

1. When there are 0 snapshots, render "No snapshots in this repository" (I'm sure Gail will suggest better copy). No link, because I think there's no point if there are no snapshots.
2. When there are >0 snapshots, render "X snapshots found" as a link.
3. In all other scenarios, render "No snapshot information".

WDYT?